### PR TITLE
People: Update the target SDK to 33

### PR DESCRIPTION
**Pull request by yaraki**
Integrate the notification runtime permission.

Change-Id: Ife1e60a3a88b40fd4cd2d3a1e8d19e429c062007

### CODE REVIEW

File: People/app/src/main/java/com/example/android/people/ui/chat/PermissionRequest.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    object Granted : PermissionStatus
+    class Denied(val shouldShowRationale: Boolean) : PermissionStatus
+}
+
+class PermissionRequest(
+    private val fragment: Fragment,
+    private val permission: String
+) {
+
+    private val launcher = fragment.registerForActivityResult(
```
Comment (alabiaga) [inline People/app/src/main/java/com/example/android/people/ui/chat/PermissionRequest.kt:38]: nitpick, but can we move this block below the declaration of the _status variable which is done below.

**Review by alabiaga — APPROVED** [review]

File: People/app/src/main/java/com/example/android/people/ui/chat/PermissionRequest.kt (hunk 1 above)
Comment (yaraki) [reply People/app/src/main/java/com/example/android/people/ui/chat/PermissionRequest.kt:46]: Done

File: People/app/src/main/java/com/example/android/people/ui/main/MainFragment.kt [hunk 2]
```diff
@@ -45,17 +52,35 @@ class MainFragment : Fragment(R.layout.main_fragment) {
...
+        }
+
+        // Deal with updates of the permission status.
+        permissionRequest.status.observe(viewLifecycleOwner) { status ->
+            when (status) {
+                // We have the permission now. Hide the permission header.
+                is PermissionStatus.Granted -> binding.contacts.adapter = contactAdapter
+                // We don't have the permission. Show the permission header.
+                is PermissionStatus.Denied -> {
+                    binding.contacts.adapter = ConcatAdapter(headerAdapter, contactAdapter)
```
Comment (dstd) [inline People/app/src/main/java/com/example/android/people/ui/main/MainFragment.kt:78]: This makes `contacts.adapter` has no stable ids

Comment (yaraki) [reply People/app/src/main/java/com/example/android/people/ui/main/MainFragment.kt:78]: That's a very good point. I'll fix it.

Comment (dstd) [reply People/app/src/main/java/com/example/android/people/ui/main/MainFragment.kt:78]: #413 🙂

Comment (yaraki) [reply People/app/src/main/java/com/example/android/people/ui/main/MainFragment.kt:78]: Merged. Thank you so much for fix it.

**Pull request merged**